lib/main.dart

```diff
@@ -49,6 +55,23 @@ class _GalleryAppState extends State<GalleryApp> {
           return Container();
         }
 
+        if (useFirebaseEmulators && !firebaseEmulatorsConfigured) {
```

Comment:
@kroikie is there a better way to run something exactly once after we have `FirebaseApp` than this boolean state thing?

Reply:
While this would likely not be an issue for us because we are not rebuilding the widget. I would recommend that we combine the "initialization" steps to include `_initialization` as well as the steps here. So `initialization` could would represent a Future that returns after both the call to `initializeApp` and emulator config is done.

We could use `Completer` to create a Future that does all the work, then pass that Future to the Future builder in build.